frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesTopology.cy.ts

```diff
@@ -466,6 +466,69 @@ describe('Pipeline topology', () => {
         .findValue()
         .contains('False');
     });
+
+    it('Test pipeline triggered run bottom drawer output', () => {
+      initIntercepts();
+      pipelineRunDetails.visit(projectId, mockRun.run_id);
+
+      pipelineRunJobDetails.findBottomDrawer().findBottomDrawerYamlTab().click();
+      pipelineRunDetails.findBottomDrawer();
+      pipelineDetails.findYamlOutput().click();
+      const pipelineDashboardCodeEditor = pipelineDetails.getPipelineDashboardCodeEditor();
+      pipelineDashboardCodeEditor.findInput().should('not.be.empty');
+    });
+  });
+
+  describe('Pipeline run Input/Output', () => {
+    beforeEach(() => {
+      initIntercepts();
+      pipelineRunDetails.visit(projectId, mockRun.run_id);
+    });
+
+    it('Test with input/output artifacts', () => {
+      pipelineRunDetails.findTaskNode('create-dataset').click();
+      pipelineRunDetails.findRightDrawer().findRightDrawerInputOutputTab().should('be.visible');
+      pipelineRunDetails.findRightDrawer().findRightDrawerInputOutputTab().click();
+      pipelineDetails.findRunTaskRightDrawer();
+      pipelineDetails
+        .findOutputArtifacts()
+        .should('contain', 'Output artifacts')
+        .should('contain', 'iris_dataset');
+
+      pipelineRunDetails.findTaskNode('normalize-dataset').click();
+      pipelineRunDetails.findRightDrawer().findRightDrawerInputOutputTab().should('be.visible');
+      pipelineDetails.findRunTaskRightDrawer();
+      pipelineDetails
+        .findInputArtifacts()
+        .should('contain', 'Input artifacts')
+        .should('contain', 'input_iris_dataset');
+    });
+  });
+
+  describe('Pipeline run Details', () => {
+    beforeEach(() => {
+      initIntercepts();
+      pipelineRunDetails.visit(projectId, mockRun.run_id);
+    });
+
+    it('Test with the details', () => {
+      pipelineRunDetails.findTaskNode('create-dataset').click();
+      pipelineRunDetails.findRightDrawer().findRightDrawerDetailsTab().should('be.visible');
```

Comment:
store this in a variable and u can reuse them 
```suggestion
      const rightDrawer=pipelineRunDetails.findRightDrawer();
      rightDrawer.findRightDrawerDetailsTab().should('be.visible');
```